dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/dataset/metadata/inputRenderer/SuggestionInputFieldRenderer.java

```diff
@@ -146,4 +149,19 @@ private Map<String, String> findFilterValues(DatasetField datasetField, Map<Stri
         return filteredValues.containsValue(StringUtils.EMPTY) ? new HashMap<>() : filteredValues;
     }
 
+    private Map<String, String> getFilterValue(DatasetField datasetField) {
+        return datasetFieldTypeToSuggestionFilterMapping
+                .entrySet()
+                .stream()
+                .filter(entry -> entry.getKey().equalsIgnoreCase(datasetField.getTypeName()))
+                .findFirst()
+                .map(entry -> {
+                    Map<String, String> filterValues = new HashMap<>();
+                    // Assuming getFieldValue() returns an Optional or similar
+                    filterValues.put(entry.getValue(), entry.getKey());
+                    return filterValues;
```

Comment:
To me the name `getFilterValue` seems a bit misleading, but tbh not sure what would be a better one. 
The main issue is that the contract between `isDependentOnSiblings`,  `getAllowedFilters` and this method is not very obvious.
For instance, I wonder if it wouldn't be clearer to have `getAllowedFilters` return a list of `FilterConfig` objects. Those filter objects would tell how a value for that filter can be retrieved, either from dataset fields or from db filter configuration.

But I feel that would require a bigger refactoring

Reply:
I've added improved comment.